### PR TITLE
BaseStringHelper::isEmpty() static method

### PR DESCRIPTION
**Pull request by GHopperMSK**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #166

### CODE REVIEW

**Comment by GHopperMSK** [comment]
It is looks like, there are some problems with test environment.

**Comment by machour** [comment]
Just restarted the CI job for this PR.
Nightly is failing because it switched to PHP 8 [a few days ago](https://twitter.com/nikita_ppv/status/1094897743594770433).

File: src/helpers/BaseStringHelper.php [hunk 1]
```diff
@@ -447,4 +447,21 @@ public static function htmlspecialchars(string $string, int $flags, string $enco
...
+     * Check if a string is empty, relying on its length. Solves unobvious behavior when `empty('0')` equal to TRUE.
+     *
+     * @param string $string
+     * @param bool $trim
+     * @return bool
+     * @see http://php.net/empty
+     */
+    public static function isEmpty(string $string, $trim = true)
+    {
+        $string = (string) $string;
```
Comment (rob006) [inline src/helpers/BaseStringHelper.php]: You already have `string` typehint in method signature, there is no need to cast it again.

**Comment by rob006** [comment]
Adding `trim` support to this method is against SRP. And without it you can replace the method body with `strcmp($string, '') === 0`.

**Comment by GHopperMSK** [comment]
> Adding `trim` support to this method is against SRP. And without it you can replace the method body with `strcmp($string, '') === 0`.

var_dump(strcmp('', '') === 0); 
sys. time: 0.067
user time: 0.009

var_dump(strlen('') === 0);
sys. time: 0.102
user time: 0.011

I think you are right.

**Comment by GHopperMSK** [comment]
One more thing I don't like in the method:
```
StringHelper::isEmpty(null);
// Argument 1 passed to yii\helpers\BaseStringHelper::isEmpty() must be of the type string, null given...
```
I don't want check if value equal to NULL every time I call `isEmpty`.

So, eventually we have:
```
public static function isEmpty(string $string = null)
{
    return strcmp($string, '') === 0;
}
```

**Comment by samdark** [comment]
See #166

**Pull request closed without merging**